testcases/kernel/controllers/pids/pids.sh

```diff
@@ -196,6 +209,118 @@ case5()
 	fi
 }
 
+case6()
+{
+	tst_res TINFO "set a limit that is smaller than current number of pids"
+	start_pids_tasks2 $max
+
+	lim=$((max - 1))
+	ROD echo $lim \> $testpath/pids.max
+
+	pids_task1 "$testpath/tasks"
+	ret=$?
+
+	if [ "$ret" -eq "2" ]; then
+		tst_res TPASS "fork failed as expected"
+	elif [ "$ret" -eq "0" ]; then
+		tst_res TFAIL "fork didn't fail despite the limit"
+	else
+		tst_res TBROK "pids_task1 failed"
+	fi
+
+	stop_pids_tasks
+}
+
+case7()
+{
+	tst_res TINFO "the number of all child cgroup tasks larger than its parent limit"
+	
+	num=10
+	lim=$((max / num))
+	for i in `seq 1 $num`; do
+		mkdir $testpath/child$i
+		ROD echo $lim \> $testpath/child$i/pids.max
+		start_pids_tasks2_path $testpath/child$i $lim
+	done
+
+	ROD echo $max \> $testpath/pids.max
+
+	pids_task1 "$testpath/tasks"
+	ret=$?
+
+	if [ "$ret" -eq "2" ]; then
+		tst_res TPASS "fork failed as expected"
+	elif [ "$ret" -eq "0" ]; then
+		tst_res TFAIL "fork didn't fail despite the limit"
+	else
+		tst_res TBROK "pids_task1 failed"
+	fi
+
+	for i in `seq 1 $num`; do
+		stop_pids_tasks_path $testpath/child$i
+		rmdir $testpath/child$i
+	done
+
+	stop_pids_tasks
+}
+
+case8()
+{
+	tst_res TINFO "set child cgroup limit smaller than its parent limit"
+	ROD echo $max \> $testpath/pids.max
+
+	mkdir $testpath/child
+
+	lim=$((max - 1))
+	ROD echo $lim \> $testpath/child/pids.max
+	tmp=$((max - 2))
+	start_pids_tasks2_path $testpath/child $tmp
+
+	pids_task1 "$testpath/child/tasks"
+	ret=$?
+
+	if [ "$ret" -eq "2" ]; then
+		tst_res TPASS "fork failed as expected"
+	elif [ "$ret" -eq "0" ]; then
+		tst_res TFAIL "fork didn't fail despite the limit"
+	else
+		tst_res TBROK "pids_task1 failed"
+	fi
+
+	stop_pids_tasks_path $testpath/child
+	rmdir $testpath/child
+	stop_pids_tasks_path
+}
+
+case9()
+{
+	tst_res TINFO "migrate cgroup"
+	mkdir $testpath/child1
+	mkdir $testpath/child2
+
+	start_pids_tasks2_path $testpath/child1 $max
+	start_pids_tasks2_path $testpath/child2 $max
+	ROD head -n 1 $testpath/child1/tasks \> $testpath/child2/tasks
+
+	if [ $(cat "$testpath/child1/pids.current") -eq $((max - 1)) ]; then
+		tst_res TPASS "migrate child1 cgroup as expected"
+	else
+		tst_res TFAIL "migrate child1 cgroup failed"
+	fi
+
+	if [ $(cat "$testpath/child2/pids.current") -eq $((max + 1)) ]; then
+		tst_res TPASS "migrate child2 cgroup as expected"
+	else
+		tst_res TFAIL "migrate child2 cgroup failed"
+	fi
+
```

Comment:
We may as well check that the pid is present in the child2 and not present in the child1 and not only compare number of pids. But in order to check anything that is pids specific (the migration case is, as far as I can tell, tested by the generic cgroup_fj tests) we should set up some limits and check that fork works in the child that we removed one task from and fails in the one that we moved the task to.

Reply:
Your proposal is great, but if a process is migrated to another cgroup, it is limited by new cgroup. Above the test has test and check that fork whether works. So I think the proposal of your last word is redundant.

Reply:
It's not really redundant, it's a simple assertion that the limit is correctly accounted for when we remove processes from the subgroup, which is something that we do not test currently.

Reply:
It make sense to me, updated.